sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -665,3 +666,16 @@ rom_error_t boot_data_check(const boot_data_t *boot_data) {
 
   return kErrorBootDataInvalid;
 }
+
+rom_error_t boot_data_as_v2(boot_data_t *boot_data) {
+  switch (launder32(boot_data->version)) {
+    case kBootDataVersion1:
+      boot_data->primary_bl0_slot = kBootDataSlotA;
+      return kErrorOk;
```

Comment:
Updating the field like this would break the digest and cause the boot_data entry to fail any subsequent validity checks.
While we can add a new getter for this field I _think_ it would be counter-intuitive since other fields don't have getters.

I think we need the following updates:
- The `version` field should also be set to `kBootDataVersion2`, and
- The `digest` field should be updated using `boot_data_digest_compute()`.

We should also make sure to call this function only for a valid boot_data entry since we don't want to waste time updating digests of invalidated entries.